src/iceberg/catalog.h

```diff
@@ -166,8 +166,7 @@ class ICEBERG_EXPORT Catalog {
   /// \param identifier a table identifier
   /// \return instance of Table implementation referred to by identifier or
   /// ErrorKind::kNoSuchTable if the table does not exist
-  virtual Result<std::shared_ptr<Table>> LoadTable(
-      const TableIdentifier& identifier) const = 0;
+  virtual Result<std::shared_ptr<Table>> LoadTable(const TableIdentifier& identifier) = 0;
```

Comment:
```suggestion
  virtual Result<std::unique_ptr<Table>> LoadTable(const TableIdentifier& identifier) = 0;
```

nit: it is better to use std::unique_ptr

Reply:
Yes, it's better.